### PR TITLE
Nix develop

### PR DESCRIPTION
**Pull request by gila**
Yes, it worked! (the PR that is)

A lot of the nix file changes are mostly reordering of some bits, nothing too extreme really. Except for the fact that now we have a .json file that contains the URI's to our sources. 

Other than that:
 - ensure we compile to Nehalem all the way down
- disable/remove
- keep CI simple: everything runs in nix-shell, no caches no containers to push before making changes etc
- reworked the bindings a little bit which is needed for the upcoming upgrade

PS Clippy might fail on an issue already fixed by, that test case will be ignored (i.e push regardless)

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -6,36 +6,23 @@ on:
...
-  Build_and_run_cargo:
+  nix-shell:
     name: Build and run cargo tests
     runs-on: self-hosted
-    timeout-minutes: 20
+    timeout-minutes: 30
     container:
-      image: docker.io/mayadata/ms-buildenv:latest
-      options: --privileged -v /dev:/dev -v /bin:/host/bin -v /lib/modules:/lib/modules
+      image: docker.io/mayadata/ms-buildenv:gila
```
Comment (tiagolobocastro) [inline .github/workflows/build.yaml:14]: Is this the correct image? (thought we'd be using the base nix image now)

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -1,22 +1,17 @@
...
 on:
   pull_request:
+  push:
 jobs:
   Clippy:
     name: Clippy
     runs-on: ubuntu-latest
     container:
-      image: docker.io/mayadata/ms-buildenv:latest
+      image: docker.io/mayadata/ms-buildenv:gila
```
Comment (tiagolobocastro) [inline .github/workflows/pr.yaml:10]: same question as the previous and also on Js checks, should we update that image too?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (gila) [reply .github/workflows/build.yaml:14]: I will change it at the end yes :)

File: nix/pkgs/mayastor/default.nix [hunk 3]
```diff
@@ -43,91 +32,54 @@ rec {
             lib.hasPrefix (toString (src + "/${allowedPrefix}")) path)
           allowedPrefixes)
       src;
+in
+rustPlatform.buildRustPackage rec {
+  name = "mayastor";
+  cargoSha256 = "1g2yq30p5z15qr94bl22bhh4p6idm5jfmh6kh88w3ixlacjyk8pf";
+  #cargoSha256 = "0000000000000000000000000000000000000000000000000000";
```
[the remark this answers is not in the document]
Comment (gila) [reply nix/pkgs/mayastor/default.nix:39]: When you need to recalculate the sha256 you have to use all zero's. so you can swap them out easily

File: scripts/node-test.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env sh
+set -euxo pipefail
+export PATH=$PATH:${HOME}/.cargo/bin
+cargo build --all
+( cd mayastor-test && npm install && ./node_modules/mocha/bin/mocha test_cli.js )
```
[the remark this answers is not in the document]
Comment (gila) [reply scripts/node-test.sh:5]: Shoot! you are right!

File: scripts/test.sh [hunk 5]
```diff
@@ -1,12 +1,11 @@
-#!/usr/bin/env bash
```
Comment (jkryl) [inline scripts/test.sh:1]: it looks like we can remove this file now as we have the test scripts under scripts/ doing the same ..

File: shell.nix [hunk 6]
```diff
@@ -1,53 +1,58 @@
...
+  channel = import ./nix/lib/rust.nix { inherit sources; };
 in
 mkShell {
+
   # fortify does not work with -O0 which is used by spdk when --enable-debug
   hardeningDisable = [ "fortify" ];
-
   buildInputs = [
-    figlet
+    #    nats-server
```
Comment (jkryl) [inline shell.nix:21]: nats-server is needed by the tests. Was it removed by accident?

File: scripts/test.sh (hunk 5 above)
Comment (gila) [reply scripts/test.sh:1]: here all tests are run, and i like to run it before I do PRs etc. Unless you have a strong objection i'd like to keep it.

File: shell.nix (hunk 6 above)
Comment (gila) [reply shell.nix]: Oh yes -- restored, tnx!

File: .github/workflows/pr.yaml [hunk 7]
```diff
@@ -1,22 +1,17 @@
 name: "Lint and style Checks"
 on:
   pull_request:
+  push:
```
[the remark this answers is not in the document]
Comment (gila) [reply .github/workflows/pr.yaml]: Yes indeed, I did not want to open/close the PR -- removed now.

**Pull request merged**